tests/test_label_categories.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+
+from datumaro import LabelCategories
+
+
+def test_add_category():
+    categories = LabelCategories()
+    index = categories.add("cat")
+    assert index == 0
+    assert len(categories) == 1
+    assert categories[0].name == "cat"
+
+
+def test_add_category_with_parent():
+    categories = LabelCategories()
+    index = categories.add("cat", parent="animal")
+    assert index == 0
+    assert len(categories) == 1
+    assert categories[0].name == "cat"
+    assert categories[0].parent == "animal"
+
+
+def test_add_category_with_attributes():
+    categories = LabelCategories()
+    attributes = {"color", "size"}
+    index = categories.add("cat", attributes=attributes)
+    assert index == 0
+    assert len(categories) == 1
+    assert categories[0].attributes == attributes
+
+
+def test_add_duplicate_category():
+    categories = LabelCategories()
+    categories.add("cat")
+    with pytest.raises(KeyError):
```

Comment:
It makes sense to check the message here, as KeyError is quite a broad thing, which can come from different sources.

Reply:
Agreed